/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/mmengine/utils/dl_utils/setup_env.py:56: UserWarning: Setting MKL_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed.
  warnings.warn(
/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/mmengine/utils/dl_utils/setup_env.py:56: UserWarning: Setting MKL_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed.
  warnings.warn(
Traceback (most recent call last):
  File "tools/train.py", line 162, in <module>
Traceback (most recent call last):
  File "tools/train.py", line 162, in <module>
        main()main()

  File "tools/train.py", line 155, in main
  File "tools/train.py", line 155, in main
        runner = Runner.from_cfg(cfg)runner = Runner.from_cfg(cfg)

  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/mmengine/runner/runner.py", line 445, in from_cfg
  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/mmengine/runner/runner.py", line 445, in from_cfg
        runner = cls(runner = cls(

  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/mmengine/runner/runner.py", line 412, in __init__
  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/mmengine/runner/runner.py", line 412, in __init__
    self.model = self.build_model(model)    self.model = self.build_model(model)
  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/mmengine/runner/runner.py", line 819, in build_model

    model = MODELS.build(model)  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/mmengine/runner/runner.py", line 819, in build_model

  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/mmengine/registry/registry.py", line 570, in build
    model = MODELS.build(model)
  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/mmengine/registry/registry.py", line 570, in build
    return self.build_func(cfg, *args, **kwargs, registry=self)    return self.build_func(cfg, *args, **kwargs, registry=self)
  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/mmengine/registry/build_functions.py", line 232, in build_model_from_cfg

    return build_from_cfg(cfg, registry, default_args)  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/mmengine/registry/build_functions.py", line 232, in build_model_from_cfg

  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/mmengine/registry/build_functions.py", line 98, in build_from_cfg
        obj_cls = registry.get(obj_type)return build_from_cfg(cfg, registry, default_args)
  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/mmengine/registry/registry.py", line 451, in get

    self.import_from_location()  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/mmengine/registry/build_functions.py", line 98, in build_from_cfg

    obj_cls = registry.get(obj_type)  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/mmengine/registry/registry.py", line 376, in import_from_location

    import_module(loc)  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/mmengine/registry/registry.py", line 451, in get

  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/importlib/__init__.py", line 127, in import_module
    self.import_from_location()    return _bootstrap._gcd_import(name[level:], package, level)
  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/mmengine/registry/registry.py", line 376, in import_from_location

    import_module(loc)  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import

  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/importlib/__init__.py", line 127, in import_module
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
    return _bootstrap._gcd_import(name[level:], package, level)  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked

  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "/home/nr4325/Desktop/Pose4/mmpose/mmpose/models/__init__.py", line 2, in <module>
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/nr4325/Desktop/Pose4/mmpose/mmpose/models/__init__.py", line 2, in <module>
    from .backbones import *  # noqa    from .backbones import *  # noqa
  File "/home/nr4325/Desktop/Pose4/mmpose/mmpose/models/backbones/__init__.py", line 37, in <module>

  File "/home/nr4325/Desktop/Pose4/mmpose/mmpose/models/backbones/__init__.py", line 37, in <module>
        from .vision_transformer import VisionTransformerfrom .vision_transformer import VisionTransformer

  File "/home/nr4325/Desktop/Pose4/mmpose/mmpose/models/backbones/vision_transformer.py", line 12, in <module>
  File "/home/nr4325/Desktop/Pose4/mmpose/mmpose/models/backbones/vision_transformer.py", line 12, in <module>
    from mmpretrain.model.utils import (MultiheadAttention, SwiGLUFFNFused, build_norm_layer,    from mmpretrain.model.utils import (MultiheadAttention, SwiGLUFFNFused, build_norm_layer,
ModuleNotFoundError
: No module named 'mmpretrain.model'ModuleNotFoundError: 
No module named 'mmpretrain.model'
/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/torch/distributed/launch.py:178: FutureWarning: The module torch.distributed.launch is deprecated
and will be removed in future. Use torchrun.
Note that --use_env is set by default in torchrun.
If your script expects `--local_rank` argument to be set, please
change it to read from `os.environ['LOCAL_RANK']` instead. See 
https://pytorch.org/docs/stable/distributed.html#launch-utility for 
further instructions

  warnings.warn(
ERROR:torch.distributed.elastic.multiprocessing.api:failed (exitcode: 1) local_rank: 0 (pid: 1098602) of binary: /home/nr4325/miniconda3/envs/pose4/bin/python
Traceback (most recent call last):
  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/torch/distributed/launch.py", line 193, in <module>
    main()
  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/torch/distributed/launch.py", line 189, in main
    launch(args)
  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/torch/distributed/launch.py", line 174, in launch
    run(args)
  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/torch/distributed/run.py", line 715, in run
    elastic_launch(
  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/torch/distributed/launcher/api.py", line 131, in __call__
    return launch_agent(self._config, self._entrypoint, list(args))
  File "/home/nr4325/miniconda3/envs/pose4/lib/python3.8/site-packages/torch/distributed/launcher/api.py", line 245, in launch_agent
    raise ChildFailedError(
torch.distributed.elastic.multiprocessing.errors.ChildFailedError: 
============================================================
tools/train.py FAILED
------------------------------------------------------------
Failures:
[1]:
  time      : 2024-12-13_04:55:42
  host      : skl-a-48.rc.rit.edu
  rank      : 1 (local_rank: 1)
  exitcode  : 1 (pid: 1098603)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
------------------------------------------------------------
Root Cause (first observed failure):
[0]:
  time      : 2024-12-13_04:55:42
  host      : skl-a-48.rc.rit.edu
  rank      : 0 (local_rank: 0)
  exitcode  : 1 (pid: 1098602)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
============================================================
